src/main/scala/com/microsoft/hyperspace/util/ResolverUtils.scala

```diff
@@ -71,4 +76,101 @@ object ResolverUtils {
+
+    /**
+     * The method fixes the casing of the components of an access field path
+     *
+     * Given a dataset with the following schema:
+     * {{{
+     * root
+     *  |-- id: integer (nullable = true)
+     *  |-- nested: struct (nullable = true)
+     *  |    |-- nested2: struct (nullable = true)
+     *  |    |    |-- v: integer (nullable = true)
+     *  |    |    |-- cnt: integer (nullable = true)
+     *  |    |    |-- other: string (nullable = true)
+     * }}}
+     * and the following parts: {{{nesTed, neSTED2, CNT}}}
+     * it will output: {{{nested, nested2, cnt}}}
+     *
+     * @param parts The parts that compose an access field path.
+     * @param outputs The output attributes of a plan.
+     * @return The parts with casing fixed as a collection.
+     */
+    def fixParts(parts: Seq[String], outputs: Seq[Attribute]): Seq[String] = {
+      var caseFixedParts = Seq.empty[String]
+      val h :: t = parts.toList
+      // Given the parts of an access field path we find the top level column
+      val topLevelFieldAttribute = outputs.find(_.name.compareToIgnoreCase(h) == 0)
+      topLevelFieldAttribute match {
+        // Top level column found
+        case Some(o) =>
+          // Exchange the given part name with the name from the toplevel output attribute
+          caseFixedParts :+= o.name
+          var children = o.dataType.asInstanceOf[StructType]
+          // With each next part we go one level deeper in the struct
+          t.foreach { e =>
+            // In all the leaves find the id of the one having the part name
+            // searching case insensitively
+            val elIdx: Int = children.fieldNames.indexWhere { f => f.compareToIgnoreCase(e) == 0 }
+            // When found add it to the new set of fixed parts
+            if (elIdx >= 0) {
+              caseFixedParts :+= children.fieldNames(elIdx)
+              children.fields.toSeq(elIdx).dataType match {
+                case s: StructType => // et it up for the next part
+                  children = s
+                case _ => // Nothing to do, cannot go deeper, it must be a leaf
+              }
+            } else {
+              caseFixedParts :+= e
+            }
+          }
+        // Top level column not found
+        case _ =>
+          // Fall back to the previously given parts
+          caseFixedParts = parts
+      }
+      caseFixedParts
+    }
+
+    val schemaFieldNames = plan.output.map(_.name)
+    Some(requiredStrings.map { requiredField =>
+      plan
+        .resolveQuoted(requiredField, spark.sessionState.conf.resolver)
+        .getOrElse { return None }
+        .collectFirst {
+          case a: AttributeReference =>
+            schemaFieldNames.find(_.compareToIgnoreCase(a.name) == 0).getOrElse(a.name)
+          case g: GetStructField =>
+            val parts = g.sql // returns backtick enclosed string (ie: `a`.`b`.`c`)
+              .replaceAll("^`(.*)`$", "$1") // strip first and last backticks
```

Comment:
How about something like the following so that we don't do regex?
```scala
  def resolve(
      spark: SparkSession,
      requiredStrings: Seq[String],
      plan: LogicalPlan): Option[Seq[String]] = {
    val schema = plan.schema
    val resolver = spark.sessionState.conf.resolver
    val resolved = requiredStrings.map { colName =>
      val resolvedColNameParts = plan
        .resolveQuoted(colName, resolver)
        .map(extractColumnName)
        .getOrElse(return None) // We can improve this if you don't want to return in the middle.
      validateResolvedColumnName(colName, resolvedColNameParts)
      getColumnNameFromSchema(schema, resolvedColNameParts, resolver).mkString(".")
    }
    Some(resolved)
  }

  private def extractColumnName(expr: Expression): Seq[String] = {
    expr match {
      case a: Attribute => Seq(a.name)
      case g @ GetStructField(_, _, Some(name)) => extractColumnName(g.child) :+ name
      case Alias(nested: GetStructField, _) => extractColumnName(nested)
    }
  }

  // Validate the resolved column name by checking if nested columns have dots in its field names.
  private def validateResolvedColumnName(
      origCol: String,
      resolvedColNameParts: Seq[String]): Unit = {
    if (resolvedColNameParts.length > 1 && resolvedColNameParts.exists(_.contains("."))) {
      throw HyperspaceException(
        s"Hyperspace does not support the nested column " +
          s"whose name contains dots: $origCol")
    }
  }

  // Given resolved column name parts, return new column name parts using the name in the given
  // schema to use the original name in order to preserve the casing.
  private def getColumnNameFromSchema(
      schema: StructType,
      resolvedColNameParts: Seq[String],
      resolver: Resolver): Seq[String] = resolvedColNameParts match {
    case h :: tail =>
      val field = schema.find(f => resolver(f.name, h)).get
      field match {
        case StructField(name, s: StructType, _, _) =>
          name +: getColumnNameFromSchema(s, tail, resolver)
        case StructField(_, _: ArrayType, _, _) =>
          throw HyperspaceException("Array types are not supported.")
        case StructField(_, _: MapType, _, _) =>
          throw HyperspaceException("Map types are not supported")
        case f => Seq(f.name)
      }
  }
```

Reply:
@imback82 I'll try out your suggestion but there are some things that are not properly working in the proposed code.

This is what I'll do:
1. Split the code in the those three private methods as they seem to be a nice grouping of actions
2. I will use `.sql` from a `GetStructField` and still fix its case because:
    1. the pattern matching fails in the case when `name` is `None` 
    2. even if the `name` has some value that needs to be case fixed
3. Use the `resolver` once more as suggested to fix the case by extracting the field access paths from schema hoping that the resolver will not keep the wrong casing.

Just for my curiosity, why against regex?

Reply:
@imback82: One more question...

When doing:
```
resolvedColNameParts.length > 1 && resolvedColNameParts.exists(_.contains("."))
```
the `> 1` is because you want to still support an index on a top level field that is not nested but contains `.` (dot)?

Reply:
> there are some things that are not properly working in the proposed code.

Oh, do you know which part? I ran `ResolverUtilsTest` and they seem fine. One case could be `GetStructField` with its `name` = `None`, but I think we can do the following if needed (but didn't check Spark code when this can be `None`):
```scala
      case g @ GetStructField(_, _, None) =>
        extractColumnName(g.child) :+ g.childSchema(g.ordinal).name
```
> the `> 1` is because you want to still support an index on a top level field that is not nested but contains `.` (dot)?

Yes, 👍🏼

> Just for my curiosity, why against regex?

In this case, it seemed we were doing `Seq("a", "b").mkString(",").split(",").map { ... }` where we could be just doing `Seq("a", "b").map { ... }`; an extreme example, but hopefully you get the idea.